geomstats/geometry/hypersphere.py

```diff
@@ -597,6 +597,33 @@ def dist(self, point_a, point_b):
 
         return dist
 
+    def dist_pairwise(self, point):
+        """Compute the pairwise distance between points.
+
+        Parameters
+        ----------
+        point : array-like, shape=[n_samples, dim]
+            Set of points belonging to the hypersphere.
+
+        Returns
+        -------
+        dist : array-like, shape=[n_samples, n_samples]
+            Pairwise distance matrix between all points.
+        """
+        n_samples, _ = point.shape
+
+        dist = gs.zeros((n_samples, n_samples))
+
+        for i in range(0, n_samples):
+            for j in range(i + 1, n_samples):
+                if i == j:
+                    continue
+
+                dist[i][j] = self.dist(point[i], point[j])
+                dist[j][i] = dist[i][j]
```

Comment:
Try to avoid assignment to make it more easily compatible with other backends, you can use append:
```python
pairwise_dist = []
for i, x in enumerate(data):
    for y in data[i:]:
        pairwise_dist.append(self.dist(x,y))
pairwise_dist = SymmetricMatrices.from_vector(pairwise_dist)
```

Reply:
Thank you, this helped me make the example work for pytorch!